lib/vdsm/supervdsm_api/udev.py

```diff
@@ -117,9 +117,9 @@ def rmAppropriateMultipathRules(thiefId):
 @expose
 def add_managed_udev_rule(vol_id, path):
     rule_file = _UDEV_RULE_FILE_NAME_MANAGED % vol_id
-    symlink = os.path.relpath(path, '/dev/')
-    rule = 'SYMLINK=="%s", RUN+="%s %s:%s $env{DEVNAME}"\n' % (
-        symlink, EXT_CHOWN, DISKIMAGE_USER, DISKIMAGE_GROUP)
+    kernel_device = os.path.relpath(path, '/dev/')
+    rule = 'KERNEL=="%s", RUN+="%s %s:%s $env{DEVNAME}"\n' % (
```

Comment:
not sure about this

Reply:
no luck with rbd:
```shell
>>> os.readlink("/run/vdsm/managedvolumes/67240876-c837-4b70-8742-604fbecabae1")
'/dev/rbd2'

>>> os.path.relpath('/dev/rbd2', '/dev/')
'rbd2'
```
```shell
$ cat /etc/udev/rules.d/99-vdsm-managed_67240876-c837-4b70-8742-604fbecabae1.rules
SYMLINK=="rbd2", RUN+="/usr/bin/chown vdsm:qemu $env{DEVNAME}"
$ udevadm trigger
$ ll /dev/rbd2
brw-rw----. 1 root disk 252, 32 Mar  3 15:12 /dev/rbd2
```

With `KERNEL`:
```shell
$ cat /etc/udev/rules.d/99-vdsm-managed_67240876-c837-4b70-8742-604fbecabae1.rules
KERNEL=="rbd2", RUN+="/usr/bin/chown vdsm:qemu $env{DEVNAME}"
$ udevadm trigger
$ ll /dev/rbd2
brw-rw----. 1 vdsm qemu 252, 32 Mar  3 15:13 /dev/rbd2
```
Needs additional research, I need to test an iscsi driver as well, only checked rbd

Reply:
Looks like `KERNEL` works for iSCSI as well (tested with kaminario as I didn't have other HW)
```shell
$ cat /etc/udev/rules.d/99-vdsm-managed_10e2fb62-7cf9-4d28-a5d8-1267cc650aba.rules
KERNEL=="dm-2", RUN+="/usr/bin/chown vdsm:qemu $env{DEVNAME}"
```